src/vcpkg/versions.cpp

```diff
@@ -265,26 +265,20 @@ namespace vcpkg::Versions
 
     VerComp compare(const std::string& a, const std::string& b, Scheme scheme)
     {
-        if (scheme == Scheme::String)
-        {
-            return (a == b) ? VerComp::eq : VerComp::unk;
-        }
-        if (scheme == Scheme::Semver)
-        {
-            return compare(semver_from_string(a).value_or_exit(VCPKG_LINE_INFO),
-                           semver_from_string(b).value_or_exit(VCPKG_LINE_INFO));
-        }
-        if (scheme == Scheme::Relaxed)
-        {
-            return compare(relaxed_from_string(a).value_or_exit(VCPKG_LINE_INFO),
-                           relaxed_from_string(b).value_or_exit(VCPKG_LINE_INFO));
-        }
-        if (scheme == Scheme::Date)
+        switch (scheme)
         {
-            return compare(DateVersion::from_string(a).value_or_exit(VCPKG_LINE_INFO),
-                           DateVersion::from_string(b).value_or_exit(VCPKG_LINE_INFO));
+            case Scheme::String: return (a == b) ? VerComp::eq : VerComp::unk;
+            case Scheme::Semver:
```

Comment:
This case is never activated in the product code -- the only caller is `Dependencies::compare_versions()`, which flattens `Scheme::Semver` to `Relaxed` first.

I'd personally inline this function and expose the individual `compare()`'s and `x_from_string()`'s, but simply removing this case and letting it hit the default unreachable would also be fine by me.

Reply:
It isn't activated in the product code right now but that doesn't seem to be a reasonable contractual assumption of this. Should I introduce a new type "DigestedVersion" or similar which is that type after flattening out semver happens?

Reply:
That would be fine by me as well.

Reply:
That resulted in going down a bit of a rabbit hole you may want to re-review.